packages/protocol/src/utils/event.ts

```diff
@@ -46,6 +46,49 @@ export interface Event<T> extends jsonrpc.Event<T> {
     (listener: (e: T) => unknown, thisArgs?: unknown, disposables?: Disposable[]): Disposable;
 }
 
+export namespace Event {
+    /**
+     * Utility function to register a one-time listener for an event. The listener will be disposed
+     * automatically after the next event is fired.
+     * @param event The event to listen to
+     * @param listener The listener function that will be called when the event happens.
+     * @param thisArgs The 'this' which will be used when calling the event listener.
+     * @param disposables An array to which the {@link Disposable} for removing the listener will be added.
+     * @returns a {@link Disposable} to remove the listener again.
+     */
+    export function once<T>(event: Event<T>, listener: (e: T) => unknown, thisArgs?: unknown, disposables?: Disposable[]): Disposable {
```

Comment:
It seems like this function is not actually used anywhere in our code. I don't mind more utility functions, juts wondering about it. Also, isn't this basically the same as the `waitUntil` without a predicate?

Reply:
Yes you are right its currently not in use.
However, I know from other projects (working with theia or vscode events) that I quite often have the use case for a self-disposing one-time listener. So having a utility function for that definitely not hurts.
I have not checked the current code base yet, so maybe we already have some places in the code where we could use this function instead of creating the listener manually.
>Also, isn't this basically the same as the waitUntil without a predicate
No, since `waitUntil` doesn`t not allow you to pass a listener function.
Also calling `waitUntil` in void fashion (i.e without awaiting the resolution) is probably an anitpattern because you end up with a floating promise.

Reply:
Fair enough, just wanted to ask.